cli/pcluster/easyconfig.py

```diff
@@ -233,11 +315,75 @@ def configure(args):  # noqa: C901 FIXME!!!
         if e.errno != errno.EEXIST:
             raise  # can safely ignore EEXISTS for this purpose...
 
+    if not _is_config_valid(args, config):
+        sys.exit(1)
+
+    # If we are here, than the file it's correct and we can override it.
     # Write configuration to disk
     open(config_file, "a").close()
     os.chmod(config_file, stat.S_IRUSR | stat.S_IWUSR)
     with open(config_file, "w") as cf:
         config.write(cf)
 
+
+def _is_config_valid(args, config):
+    """
+    Validate the configuration of the pcluster configure.
+
+    :param args: the arguments passed with the command line
+    :param config: the configParser
+    :return True if the configuration is valid, false otherwise
+    """
+    # We create a temp_file_path to validate before overriding the original config
+    temp_file_path = os.path.join(tempfile.gettempdir(), "temp_config")
+    temp_args = copy.copy(args)  # Defensive copy is needed because we change config_file
+
+    temp_args.config_file = temp_file_path
+    with open(temp_file_path, "w+") as cf:
+        config.write(cf)
     # Verify the configuration
-    cfnconfig.ParallelClusterConfig(args)
+    is_valid = True
+    try:
+        cfnconfig.ParallelClusterConfig(temp_args)
+    except SystemExit:
+        is_valid = False
+    finally:
+        os.remove(temp_file_path)
+        return is_valid
+
+
+def get_config_parameter(config, section, parameter_name, default_value):
+    """
+    Get the parameter if present in the configuration otherwise returns default value.
+
+    :param config the configuration parser
+    :param section the name of the section
+    :param parameter_name: the name of the parameter
+    :param default_value: the default to propose the user
+    :return:
+    """
+    return config.get(section, parameter_name) if config.has_option(section, parameter_name) else default_value
+
+
+def scheduler_handler(scheduler):
+    """
+    Return a dictionary containing information based on the scheduler.
+
+    :param scheduler the target scheduler
+    :return: a dictionary with containing the information
+    """
+    scheduler_info = {}
+    if scheduler == "awsbatch":
+        scheduler_info["parameters_to_remove"] = (
+            "max_queue_size",
+            "initial_queue_size",
+            "maintain_initial_size",
+            "compute_instance_type",
+        )
+        scheduler_info["max_size_param"] = "max_vcpus"
+        scheduler_info["initial_cpus"] = "desired_vcpus"
+    else:
+        scheduler_info["parameters_to_remove"] = "max_vcpus", "desired_vcpus", "min_vcpus", "compute_instance_type"
+        scheduler_info["max_size_param"] = "max_queue_size"
+        scheduler_info["initial_cpus"] = "initial_queue_size"
```

Comment:
I'd call "initial_cluster_size" or "desired_size" because it can be initial_queue_size or max_vcpus